src/Microsoft.Health.Fhir.Shared.Core/Features/Search/RawBundleEntryComponent.cs

```diff
@@ -15,11 +15,14 @@ namespace Microsoft.Health.Fhir.Shared.Core.Features.Search
     [FhirType("EntryComponent")]
     public class RawBundleEntryComponent : Bundle.EntryComponent
     {
+        private readonly ResourceWrapper _resourceWrapper;
+
         public RawBundleEntryComponent(ResourceWrapper resourceWrapper)
         {
             EnsureArg.IsNotNull(resourceWrapper, nameof(resourceWrapper));
 
             ResourceElement = new RawResourceElement(resourceWrapper);
+            _resourceWrapper = resourceWrapper;
```

Comment:
`_resourceWrapper` and `ResourceElement` represent the same object but `ResourceElement` isn't immutable. Are there any references where these could get out of sync?

Reply:
Fixed by changing to save just the isDeleted status.